beacon-chain/rpc/eth/light-client/handlers.go

```diff
@@ -0,0 +1,402 @@
+func (bs *Server) GetLightClientOptimisticUpdate(w http.ResponseWriter, req *http.Request) {
+	// Prepare
+	ctx, span := trace.StartSpan(req.Context(), "beacon.GetLightClientOptimisticUpdate")
+	defer span.End()
+
+	minSignatures := params.BeaconConfig().MinSyncCommitteeParticipants
+
+	block, err := bs.getLightClientEventBlock(ctx, minSignatures)
+	if !shared.WriteBlockFetchError(w, block, err) {
+		return
+	}
+
+	state, err := bs.Stater.StateBySlot(ctx, block.Block().Slot())
+	if err != nil {
+		http2.HandleError(w, "could not get state "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// Get attested state
+	attestedRoot := block.Block().ParentRoot()
+	attestedBlock, err := bs.BeaconDB.Block(ctx, attestedRoot)
+	if err != nil {
+		http2.HandleError(w, "could not get attested block "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if attestedBlock == nil {
+		http2.HandleError(w, "attested block is nil", http.StatusInternalServerError)
+		return
+	}
+
+	attestedSlot := attestedBlock.Block().Slot()
+	attestedState, err := bs.Stater.StateBySlot(ctx, attestedSlot)
+	if err != nil {
+		http2.HandleError(w, "could not get attested state "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	update, err := NewLightClientOptimisticUpdateFromBeaconState(
+		ctx,
+		state,
+		block,
+		attestedState,
+	)
+	if err != nil {
+		http2.HandleError(w, "could not get light client optimistic update "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	response := &LightClientUpdateWithVersion{
+		Version: ethpbv2.Version(attestedState.Version()).String(),
+		Data:    update,
+	}
+
+	http2.WriteJson(w, response)
+}
+
+// getLightClientEventBlock - returns the block that should be used for light client events, which satisfies the minimum number of signatures from sync committee
+func (bs *Server) getLightClientEventBlock(ctx context.Context, minSignaturesRequired uint64) (interfaces.ReadOnlySignedBeaconBlock, error) {
+	// Get the current state
+	state, err := bs.HeadFetcher.HeadState(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("could not get head state %v", err)
+	}
+
+	// Get the block
+	latestBlockHeader := *state.LatestBlockHeader()
+	stateRoot, err := state.HashTreeRoot(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("could not get state root %v", err)
+	}
+	latestBlockHeader.StateRoot = stateRoot[:]
+	latestBlockHeaderRoot, err := latestBlockHeader.HashTreeRoot()
+	if err != nil {
+		return nil, fmt.Errorf("could not get latest block header root %v", err)
+	}
+
+	block, err := bs.BeaconDB.Block(ctx, latestBlockHeaderRoot)
+	if err != nil {
+		return nil, fmt.Errorf("could not get latest block %v", err)
+	}
+	if block == nil {
+		return nil, fmt.Errorf("latest block is nil")
+	}
+	// Loop through the blocks until we find a block that has super majority of sync committee signatures (2/3)
```

Comment:
Are you sure this comment is correct? Isn't supermajority required only when calling from `GetLightClientFinalityUpdate`?

Reply:
Good catch, the comment isn't correct. 

Patched.